pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -565,6 +565,10 @@ type Template struct {
 	// pods created by those templates will not be counted towards this total.
 	Parallelism *int64 `json:"parallelism,omitempty" protobuf:"bytes,23,opt,name=parallelism"`
 
+	// MaxFailed, if specified, is the maximum number of failed child pods that are allowed before the template itself
+	// stopped and marked as failed
+	MaxFailed *int64 `json:"maxFailed,omitempty" protobuf:"varint,41,opt,name=maxFailed"`
+
```

Comment:
This spec is certainly not set in stone and would like feedback on this.

Mainly this feels constrained... if we're adding a whole new field should we allow expressions here? This way we could consider Skipped, Errored, etc. as well as Failed steps

Reply:
minor, I'm assuming you're not using `uint32` because of some GRPC stuff?

Reply:
should this be under something like `retry: `?

Reply:
`FailFast *bool` for now and create a `.next` enhancement proposal